### PR TITLE
Fix tag createWithSignature function

### PR DESCRIPTION
**Pull request by julianmesa-gitkraken**
Fix tag createWithSignature function
Also tests fixed

### CODE REVIEW

File: lib/tag.js [hunk 1]
```diff
@@ -71,7 +71,7 @@ Tag.createWithSignature = function(
   signingCallback
 ) {
   let tagBuffer;
-  return Tag.createBuffer(repo, tagName, target, tagger, message)
+  return Tag.createBuffer(repo, tagName, target.id(), tagger, message)
```
Comment (ianhattendorf) [inline lib/tag.js:74]: Can we check if `target.id` exists, and if not throw a deprecation warning while still functioning? That way we give users a heads up that they're using the API incorrectly without abruptly breaking things for them. Something like (pseudocode, not tested)
```
const deprecatedCreateWithSignatureHelper = util.deprecate(function(oidTarget) {
  return repo.getCommit(oidTarget);
}, "Tag.createWithSignature target should be a Git Object, not Oid");

Tag.createWithSignature = async function(
  ...


  if (!target.id) {
   target = await deprecatedCreateWithSignatureHelper(target);
  }

...  
```

See https://nodejs.org/api/util.html#utildeprecatefn-msg-code if you aren't familiar with `util.deprecate`, we use this elsewhere in NodeGit.

**Review by ianhattendorf — CHANGES_REQUESTED** [review]

File: lib/tag.js (hunk 1 above)
Comment (julianmesa-gitkraken) [reply lib/tag.js]: Done.

**Comment by julianmesa-gitkraken** [comment]
@ianhattendorf Changed `createWithSignature` using async/await.
Also added a test to show the deprecation warning.

**Review by ianhattendorf — APPROVED** [review]
Thanks for adding the deprecation message :+1:

**Pull request merged**